tests/pipelines/test_pipelines_image2text_generation.py

```diff
@@ -0,0 +1,119 @@
+
+
+if is_vision_available():
+    from PIL import Image
+else:
+
+    class Image:
+        @staticmethod
+        def open(*args, **kwargs):
+            pass
+
+
+@is_pipeline_test
+@require_torch
+@require_vision
+class Image2TextGenerationPipelineTests(unittest.TestCase, metaclass=PipelineTestCaseMeta):
+    model_mapping = MODEL_FOR_VISION_2_SEQ_MAPPING
+
+    def get_test_pipeline(self, model, tokenizer, feature_extractor):
+        pipe = pipeline("image2text-generation", model="nlpconnect/vit-gpt2-image-captioning")
+        examples = [
+            Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png"),
+            "./tests/fixtures/tests_samples/COCO/000000039769.png",
+        ]
+        return pipe, examples
+
+    def run_pipeline_test(self, pipe, examples):
+        outputs = pipe(examples)
+        self.assertEqual(
+            outputs,
+            [
+                [{"generated_text": ANY(str)}],
+                [{"generated_text": ANY(str)}],
+            ],
+        )
+
+    @require_flax
+    def test_small_model_flax(self):
+        pipe = pipeline("image2text-generation", model="hf-internal-testing/tiny-random-vit-gpt2")
+        image = "./tests/fixtures/tests_samples/COCO/000000039769.png"
+
+        outputs = pipe(image)
+        self.assertEqual(
+            outputs,
+            [
+                {"generated_text": ANY(str)},
+            ],
+        )
+
+        outputs = pipe([image, image])
+        self.assertEqual(
+            outputs,
+            [
+                [{"generated_text": ANY(str)}],
+                [{"generated_text": ANY(str)}],
+            ],
+        )
+
+    @require_tf
+    @unittest.skip("No small model yet on the hub")
+    def test_small_model_tf(self):
+        raise NotImplementedError
+
+    @require_torch
+    def test_small_model_pt(self):
+        pipe = pipeline("image2text-generation", model="hf-internal-testing/tiny-random-vit-gpt2")
+        image = "./tests/fixtures/tests_samples/COCO/000000039769.png"
+
+        outputs = pipe(image)
+        self.assertEqual(
+            outputs,
+            [
+                {"generated_text": ANY(str)},
+            ],
+        )
+
+        outputs = pipe([image, image])
+        self.assertEqual(
+            outputs,
+            [
+                [{"generated_text": ANY(str)}],
+                [{"generated_text": ANY(str)}],
+            ],
+        )
+
+    @slow
+    @require_torch
+    def test_large_model_pt(self):
+        pipe = pipeline("image2text-generation", model="nlpconnect/vit-gpt2-image-captioning")
+        image = "./tests/fixtures/tests_samples/COCO/000000039769.png"
+
+        outputs = pipe(image)
+        self.assertEqual(outputs, [{"generated_text": "a cat laying on a blanket next to a cat laying on a bed "}])
+
+        outputs = pipe([image, image])
+        self.assertEqual(
+            outputs, [[{"generated_text": "a cat laying on a blanket next to a cat laying on a bed "}]] * 2
```

Comment:
```suggestion
            outputs, [[{"generated_text": "a cat laying on a blanket next to a cat laying on a bed "}], [{"generated_text": "a cat laying on a blanket next to a cat laying on a bed "}]]
```
I tend to prefer overly specific in tests as it reduces strain on the reader to know if the test is ok or not.
Also it would be better if we had 2 different images to check against ,but we don't have in fixtures (we're mindful of the repo size) so this is OK like this.